deepchecks/tabular/checks/performance/model_error_analysis.py

```diff
@@ -121,10 +121,9 @@ def run_logic(self, context: Context) -> CheckResult:
             def scoring_func(dataset: Dataset):
                 return per_sample_mse(dataset.label_col, model.predict(dataset.features_columns))
         else:
-            le = preprocessing.LabelEncoder()
-            le.fit(train_dataset.classes)
-
             def scoring_func(dataset: Dataset):
+                le = preprocessing.LabelEncoder()
+                le.fit(dataset.classes)
```

Comment:
Why?

Anyway - is this scoring_func is being called only once? if not, this change is incorrect.

Reply:
it is only once
again, test and train data can have different classes